File: Dot examples/ASP Book Examples/Book Examples/Chapter27/VB/App_Code/QueryStringAuthenticationModule.vb
Imports System
Imports System.Web

Namespace AspNetUnleashed
    Public Class QueryStringAuthenticationModule
        Implements IHttpModule

        Public Sub Init(ByVal app As HttpApplication) Implements IHttpModule.Init
            AddHandler app.AuthorizeRequest, AddressOf AuthorizeRequest
        End Sub

        Private Sub AuthorizeRequest(ByVal sender As Object, ByVal e As EventArgs)
            ' Get context
            Dim app As HttpApplication = CType(sender, HttpApplication)
            Dim context As HttpContext = app.Context

            ' If the request is for Login.aspx, exit
            Dim path As String = context.Request.AppRelativeCurrentExecutionFilePath
            If String.Compare(path, "~/login.aspx", True) = 0 Then
                Return
            End If

            ' Check for password
            Dim authenticated As Boolean = False
            If Not IsNothing(context.Request.QueryString("password")) Then
                If context.Request.QueryString("password") = "secret" Then
                    authenticated = True
                End If
            End If

            ' If not authenticated, redirect to login.aspx
            If Not authenticated Then
                context.Response.Redirect("~/Login.aspx")
            End If
        End Sub

        Public Sub Dispose() Implements IHttpModule.Dispose
        End Sub
    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter27/CS/App_Code/QueryStringAuthenticationModule.cs
using System;
using System.Web;

namespace AspNetUnleashed
{
    public class QueryStringAuthenticationModule : IHttpModule
    {
        public void Init(HttpApplication app)
        {
            app.AuthorizeRequest += new EventHandler(AuthorizeRequest);
        }

        private void AuthorizeRequest(Object sender, EventArgs e)
        {
            // Get context
            HttpApplication app = (HttpApplication)sender;
            HttpContext context = app.Context;

            // If the request is for Login.aspx, exit
            string path = context.Request.AppRelativeCurrentExecutionFilePath;
            if (String.Compare(path, "~/login.aspx", true) == 0)
                return;

            // Check for password
            bool authenticated = false;
            if (context.Request.QueryString["password"] != null)
            {
                if (context.Request.QueryString["password"] == "secret")
                    authenticated = true;
            }

            // If not authenticated, redirect to login.aspx
            if (!authenticated)
                context.Response.Redirect("~/Login.aspx");
        }

        public void Dispose() { }
    }
}
